File: app/src/main/java/com/imangazalievm/bubbble/domain/shotzoom/ShotZoomInteractor.java
package com.imangazalievm.bubbble.domain.shotzoom;


import com.imangazalievm.bubbble.data.shots.ImagesRepository;
import com.imangazalievm.bubbble.presentation.global.SchedulersProvider;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ShotZoomInteractor {

    private ImagesRepository imagesRepository;
    private SchedulersProvider schedulersProvider;

    @Inject
    public ShotZoomInteractor(
            ImagesRepository imagesRepository,
            SchedulersProvider schedulersProvider
    ) {
        this.imagesRepository = imagesRepository;
        this.schedulersProvider = schedulersProvider;
    }

    public Completable saveImage(String shotImageUrl) {
        return imagesRepository.saveImage(shotImageUrl)
                .subscribeOn(schedulersProvider.io());
    }

}

File: app/src/main/java/com/imangazalievm/bubbble/domain/shotzoom/ShotZoomInteractor.kt
package com.imangazalievm.bubbble.domain.shotzoom

import com.imangazalievm.bubbble.data.shots.ImagesRepository
import com.imangazalievm.bubbble.presentation.global.SchedulersProvider
import io.reactivex.Completable
import javax.inject.Inject

class ShotZoomInteractor @Inject constructor(
    private val imagesRepository: ImagesRepository
) {

    suspend fun saveImage(shotImageUrl: String) {
        imagesRepository.saveImage(shotImageUrl)
    }

}